src/main/java/com/exasol/dbbuilder/dialects/exasol/AdapterScript.java

```diff
@@ -27,6 +30,33 @@ public AdapterScript(final ExasolImmediateDatabaseObjectWriter writer, final Sch
         this.writer = writer;
         this.language = language;
         this.content = content;
+        this.debuggerConnection = Optional.empty();
+        this.writer.write(this);
+        if (isAdapterScriptDebuggingEnabled()) {
+            throw new IllegalStateException("Adapter script debugging was enabled but no debugger address was passed."
+                    + "Please use a different constructor.");
+        }
+    }
+
+    /**
+     * Create a new instance of an {@link AdapterScript} with possible debugger connection. The debugger is only
+     * attached if the property test.debugAdapterScripts="true" is set. You can set it by appending -Dtest.debug="true"
+     * to your the JVM.
+     *
+     * @param writer             database object writer
+     * @param parentSchema       parent schema
+     * @param name               name of the adapter script
+     * @param language           language the the script is implemented in
+     * @param content            the actual script
+     * @param debuggerConnection connection string for the debugger
+     */
+    public AdapterScript(final ExasolImmediateDatabaseObjectWriter writer, final Schema parentSchema, final String name,
```

Comment:
We now reached a point, where a builder is necessary. Five parameters were too many already and it does not get better. Since the constructor should never be called directly (but from the factory) you can replace the existing constructor too.